### PR TITLE
Updates for JDK 11

### PR DESCRIPTION
**Pull request by jasonkatonica**
Modifications were made to support a Java 11 environment. This includes the following updates as compared to Java 17:

- The `mvn` configuration was modified to enforce Java 11 usage on this
branch of the code.
- The algortihms `Ed25519` and `Ed448` are lacking framework support in
Java 11. These algorithms and associated tests were removed.
- Construction of various key types has changed. The Java 11 frameworks
encoding and decoding methods needed to be called differently.
- The `java.nio.ByteBuffer` class needed adjustments to get/put bytes
into a specific position in the buffer. The existing method for Java 17
does not exist on Java 11.
- The `jgskit_resource.rc` file was modified appropriately for Java 11.
- The test `BaseTestXDHKeyImport.java` was modifed to expect a different
known answer value. There is a subtle difference between Java 11 and
Java 17 in terms of how the XDH DER encodings are represented. This
update describes the differences.
- Updated the github actions workflow to use Java 11.

Signed-off-by: Jason Katonica <katonica@us.ibm.com>

### CODE REVIEW

File: src/test/java/ibm/jceplus/junit/base/BaseTestXDHKeyImport.java [hunk 1]
```diff
@@ -125,18 +125,18 @@ public void createKeyPairXDHParamImport(String alg) throws Exception {
                     "515095759487624245475052955143775821008860566299846242602604658885545774489");
             scalar = Base64.getDecoder().decode("RtcJeC6VA0w1s2ly58eimwgmjNnIvdBQ0WwXodcOKx0=");
             actualPbk = Base64.getDecoder()
-                    .decode("MCowBQYDK2VuAyEAmU2lHYDsKSa7YoPkUeOxazej9sWb9m2hYbHEl6uIIwE=");
+                    .decode("MCwwBwYDK2VuBQADIQCZTaUdgOwpJrtig+RR47FrN6P2xZv2baFhscSXq4gjAQ==");
```
Comment (KostasTsiounis) [inline src/test/java/ibm/jceplus/junit/base/BaseTestXDHKeyImport.java:128]: Should we change all versions to this new value?

**Review by KostasTsiounis — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -234,9 +234,6 @@ KeyAgreement                | XDH                        |                 |X
 KeyFactory                  | DSA                        |X                |X             |
 KeyFactory                  | DiffieHellman              |X                |X             |
 KeyFactory                  | EC                         |X                |X             |
-KeyFactory                  | Ed25519                    |                 |X             |
```
[the remark this answers is not in the document]
Comment (jasonkatonica) [reply README.md:237]: I believe it to be cleaner to only have a row for algorithms that are supported in either `OpenJCEPlus` or `OpenJCEPlusFIPS`. If no support for a given algorithm then we would leave that out of the documentation entirely.

File: src/test/java/ibm/jceplus/junit/base/BaseTestXDHKeyImport.java (hunk 1 above)
Comment (jasonkatonica) [reply src/test/java/ibm/jceplus/junit/base/BaseTestXDHKeyImport.java:172]: I updated the Javadoc associated with this change to document what the difference is. I will also plan on making a very similar update to other releases if this looks ok to you.

Comment (KostasTsiounis) [reply src/test/java/ibm/jceplus/junit/base/BaseTestXDHKeyImport.java:172]: Makes sense to me.

**Review by KostasTsiounis — APPROVED** [review]
LGTM

**Review by JinhangZhang — APPROVED** [review]

**Comment by JinhangZhang** [comment]
LGTM

**Pull request merged**